packages/language-service/src/lib-new/features/ls-st-import.ts

```diff
@@ -50,10 +56,275 @@ function getStImportCompletions(context: LangServiceContext, _importNode: postcs
+                            originSymbol: originResolve.symbol,
+                            originPath: originResolve.meta.source,
+                        };
+                    },
+                });
+                if (importType === 'top') {
+                    addNamedImportCompletion({
+                        context,
+                        completions,
+                        availableImports: TYPED_NAMED,
+                        existingNames: existingCalls,
+                        nameBeforeCaret,
+                        createCompletion: stImportNamedCompletion.typeAssertCall,
+                    });
+                }
+            } else {
+                // JS import completions
+                addNamedImportCompletion({
+                    context,
+                    completions,
+                    availableImports: importFrom.value,
+                    existingNames,
+                    nameBeforeCaret,
+                    resolveOrigin(symbol) {
+                        return {
+                            jsValue: symbol,
+                            originPath: importFrom.resolvedPath,
+                        };
+                    },
+                });
+            }
         }
     }
     return completions;
 }
+function addNamedImportCompletion({
+    context,
+    completions,
+    availableImports,
+    existingNames,
+    nameBeforeCaret,
+    resolveOrigin,
+    createCompletion = stImportNamedCompletion,
+}: {
+    context: LangServiceContext;
+    completions: Completion[];
+    availableImports: Record<string, any>;
+    existingNames: Set<string>;
+    nameBeforeCaret: string;
+    resolveOrigin?: (symbol: StylableSymbol) => {
+        originSymbol?: StylableSymbol | undefined;
+        jsValue?: any;
+        originPath: string;
+    };
+    createCompletion?: (options: Parameters<typeof stImportNamedCompletion>[0]) => Completion;
+}) {
+    for (const [name, value] of Object.entries(availableImports)) {
+        if (existingNames.has(name)) {
+            continue;
+        }
+        let deltaStart = 0;
+        if (name.startsWith(nameBeforeCaret)) {
+            deltaStart = -nameBeforeCaret.length;
+        } else {
+            continue;
+        }
+        const originSymbolOrValue = resolveOrigin?.(value);
+        let relativePath = path.relative(
+            path.dirname(context.meta.source),
+            originSymbolOrValue?.originPath || ''
+        );
+        if (!path.isAbsolute(relativePath) && !relativePath.match(/^\./)) {
+            relativePath = './' + relativePath;
+        }
+        relativePath = relativePath.replace(/\\/g, '/');
+        completions.push(
+            createCompletion({
+                ...originSymbolOrValue,
+                localName: name,
+                rng: range(context.getPosition(), { deltaStart }),
+                relativePath,
+            })
+        );
+    }
+}
+function getSpecifierModule(context: LangServiceContext) {
```

Comment:
we have import parser. but since we already parsed it maybe it's fine?

Reply:
looking deeper seems like you need positions so the import parser is not good here